### DISCUSSION TITLE
Feature request - Road Tool

### DISCUSSION THREAD

**Discussion by raldor123** [spimort/TerraBrush#143] [Ideas]
First, I’d like to compliment this wonderful work and thank you for providing this plugin to the community. I was thinking that a great feature would be a road tool. It would be similar to what we have in Unreal Engine 5, where there is a button to create a spline that generates a flat mesh following the spline, and this mesh can be placed and manipulated over the terrain. After placing the mesh on the terrain, there would be a button to adapt the terrain to the road mesh, deforming the terrain based on the spline.

I think it would be a great idea for those wanting to make a racing game or custom roads throughout the map. I believe that simply painting these roads would be a lot more work and much more difficult. With this tool, I think it would greatly simplify the process.

If I’m not being clear, here’s a link to a video showing an example: https://www.youtube.com/watch?v=jF6K3BkAa9M&t

**Comment by spimort** [comment]
Ok wow this video is super cool!

It would be a great feature, I agree! 🙂
I think it would be the same request as this one : 
https://github.com/spimort/TerraBrush/issues/110

I'll copy your video link to the other issue!

Do you mind if we close this one in favor of #110 ?

**Reply by raldor123** [reply]
Sure, no problem! Thank you for the response. And I apologize for the mistake, I hadn't noticed that it was the same idea.

**Reply by spimort** [reply]
No problem! 🙂
